invenio_files_rest/models.py

```diff
@@ -1020,11 +1049,35 @@ def copy(self, bucket=None, key=None):
             Default: current object key.
         :returns: The copied object version.
         """
-        return ObjectVersion.create(
-            self.bucket if bucket is None else as_bucket(bucket),
-            key or self.key,
-            _file_id=self.file_id
-        )
+        # get (if exists) the latest version
```

Comment:
I think the ``copy()`` method will need some refactoring. As it is now, it will execute quite a lot of extra queries compared to the previous version. It seems the extra check to not copy an object if it already exists, was done because of the merge method. If so, perhaps the complexity can be moved to the merge method. Alternatively, we could add e.g. two parameters ``copy(self ,..., force=True, with_tags=False)`` that would control if 1) an object is copied if it already exists and 2) if tags are copied.

Reply:
Agree, the behaviour has been changed here: before, a copy was created in any case in `create()` method (if the same object version was already there, a new identical version was added as head), while now only if not already present.
Not sure an extra param is what we need. I have the feeling some sort of optimization code was added... Shall we decide what is best?